src/Xamarin.Android.Build.Tasks/Utilities/NdkTools/NoClang.cs

```diff
@@ -0,0 +1,99 @@
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+
+using Microsoft.Build.Utilities;
+using Xamarin.Android.Tools;
+
+namespace Xamarin.Android.Tasks
+{
+	abstract class NdkToolsNoClang : NdkTools
+	{
+		protected NdkToolsNoClang (string androidNdkPath, NdkVersion version, TaskLoggingHelper? log)
+			: base (androidNdkPath, version, log)
+		{
+			NdkToolNames[NdkToolKind.CompilerC] = "gcc";
+			NdkToolNames[NdkToolKind.CompilerCPlusPlus] = "g++";
+		}
+
+		public override bool ValidateNdkPlatform (Action<string> logMessage, Action<string, string> logError, AndroidTargetArch arch, bool enableLLVM)
+		{
+			// Check that we have a compatible NDK version for the targeted ABIs.
+			if (IsNdk64BitArch (arch) && Version.Main.Major < 10) {
+				logMessage (
+					"The detected Android NDK version is incompatible with the targeted 64-bit architecture, " +
+					"please upgrade to NDK r10 or newer.");
+			}
+
+			// NDK r10d is buggy and cannot link x86_64 ABI shared libraries because they are 32-bits.
```

Comment:
If NDK r14 is our minimum supported version, do we need to care about NDK r10?  We could probably just remove this code entirely.

Reply:
I think we should still keep the validation code, should anyone have NDK r10 installed. I will update the message to recommend at least r14 though

Reply:
Sorry; I misread this: https://github.com/xamarin/xamarin-android/pull/6083/files#diff-7c0f17654397609ff55091f911023c78c6325652f1c4839fa8a9cdead0772bf4R77

I thought ^^ was an "early error check + return", but it isn't.